firestore/google/cloud/firestore_v1beta1/_helpers.py

```diff
@@ -837,6 +839,34 @@ def pbs_for_set(document_path, document_data, option):
     return write_pbs
 
 
+def canonical_strings(field_paths):
+    """ Converts simple field path with integer beginnings to quoted field path
+
+    Args:
+        field_paths (Sequence[str, ...]): A list of field paths
+
+    Returns:
+        canonical_strings (Sequence[str, ...]):
+            The same list of field paths except non-simple field names
+            in the `.` delimited field path have been converted
+            into quoted unicode field paths. Simple field paths match
+            the regex ^[_a-zA-Z][_a-zA-Z0-9]*$.  See `Document`_ page for
+            more information.
+
+    .. _Document: https://cloud.google.com/firestore/docs/reference/rpc/google.firestore.v1beta1#google.firestore.v1beta1.Document  # NOQA
+    """
+    canonical_strings = []
+    for field_path in field_paths:
+        field_names = field_path.split('.')
+        escaped_names = [u"`{}`".format(
```

Comment:
This would actually be easier to read if you broke it down into a for statement with an if on the match.